src/Components/ADT3DSceneBuilder/Internal/Behaviors/Widgets/WidgetBuilders/DataHistoryWidgetBuilder/DataHistoryWidgetBuilder.tsx

```diff
@@ -121,7 +122,8 @@ const DataHistoryWidgetBuilder: React.FC<IDataHistoryWidgetBuilderProps> = ({
             const connection = adxConnectionInformation.connection;
             updateWidgetData(
                 produce(formData, (draft) => {
-                    draft.widgetConfiguration.connection = {
+                    draft.widgetConfiguration.connection = formData
+                        .widgetConfiguration.connection ?? {
```

Comment:
not to override existing widget connection information with the adt instance's connection information

Reply:
So if you configured the widget before and now come back to the form, we want to prioritize the previously stored data over some other source? What is that other source?

Reply:
yes, the connection information in the 3d scene config should be the one that we see in the form even if it is not the same as the one of the selected adt instance's data history connection information. It was a bug on our side that I just noticed. The other source is the adt instance's data history connection.

It is uncommon that users would use different 3d scene config files which includes connection information for their widgets which are not the same as their selected adt instance's adx connection information. But since I used a different one when I was testing the customer incident, I noticed that bug. We were overriding the connection string whenever we update a data history widget with the selected adt instance's connection information without relying on existing connection information in the config file.